spec/models/comment_spec.rb

```diff
@@ -31,143 +31,207 @@
+      subject { build(:comment, pseud: user.default_pseud) }
 
-      it "allows editing the comment if it's not changed significantly" do
-        subject.edited_at = Time.current
-        subject.comment_content += "a"
-        expect(subject.save).to be_truthy
-      end
+      context "spam flagging on creation" do
+        let(:user) { create(:user) }
 
-      it "allows modifying the comment besides content" do
-        subject.hidden_by_admin = true
-        expect(subject.save).to be_truthy
-      end
-    end
+        context "when spamcheck is skipped and comment is spammy" do
+          before do
+            allow_any_instance_of(Comment).to receive(:skip_spamcheck?).and_return(true)
+            allow_any_instance_of(Comment).to receive(:spam?).and_return(true)
+          end
 
-    shared_examples "always allows changing the comment" do
-      it "allows editing the comment if it's changed significantly" do
-        subject.edited_at = Time.current
-        subject.comment_content = "Spam content" * 12
-        expect(subject.save).to be_truthy
-      end
+          it "is flagged as ham" do
+            subject.save!
+            subject.reload
+            expect(subject.approved).to be_truthy
+            expect(subject.spam).to be_falsey
+          end
+        end
 
-      it "allows editing the comment if it's not changed significantly" do
-        subject.edited_at = Time.current
-        subject.comment_content += "a"
-        expect(subject.save).to be_truthy
-      end
-    end
+        context "when spamcheck is skipped and comment is not spammy" do
+          before do
+            allow_any_instance_of(Comment).to receive(:skip_spamcheck?).and_return(true)
+            allow_any_instance_of(Comment).to receive(:spam?).and_return(false)
+          end
 
-    context "when any comment is considered spam" do
-      subject { build(:comment, pseud: user.default_pseud) }
-      let(:admin_setting) { AdminSetting.first || AdminSetting.create }
+          it "is flagged as ham" do
+            subject.save!
+            subject.reload
+            expect(subject.approved).to be_truthy
+            expect(subject.spam).to be_falsey
+          end
+        end
 
-      before do
-        subject.save!
-        allow_any_instance_of(Comment).to receive(:spam?).and_return(true)
-      end
+        context "when spamcheck is not skipped and content is spammy" do
+          before do
+            allow_any_instance_of(Comment).to receive(:skip_spamcheck?).and_return(false)
+            allow_any_instance_of(Comment).to receive(:spam?).and_return(true)
+          end
 
-      context "when account_age_threshold_for_comment_spam_check is set" do
-        before do
-          admin_setting.update_attribute(:account_age_threshold_for_comment_spam_check, 10)
+          it "is flagged as spam" do
+            expect { subject.save! }
+              .to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
I think this test will be slightly faster/cleaner if you call `save` without the bang and then check the return value instead